src/MongoDB/Connection.php

```diff
@@ -95,15 +95,14 @@ public function disconnect()
     }
 
     /**
-     * @param string $aggregate - JSON string with 'pipeline' and 'options' keys
+     * @param array $aggregate - array with 'pipeline' and 'options' keys
      * @param array $bindings
      * @param bool $useReadOnly
      * @return \Generator|\Traversable
      */
     public function cursor($aggregate, $bindings = [], $useReadOnly = true)
     {
-        $args = array_merge(['pipeline' => [], 'options' => []], json_decode($aggregate, true));
-        $cursor = $this->collection->aggregate($args['pipeline'], $args['options']);
+        $cursor = $this->collection->aggregate($aggregate['pipeline'], $aggregate['options']);
```

Comment:
if either of these are missing, you'll get an undefined index error. Either put the merge back in or gracefully handle missing, please.